File: OtherLanguages/PHP/CrashReporter/Data/Application.php
<?
class Application
{
	public $id;
	public $guid;
	public $name;

	public function read($reader)
	{
		$this->id = $reader->hasField(0) ? $reader->readInteger(0) : null;
		$this->guid = $reader->readString(1);
		$this->name = $reader->readString(2);
	}

	public function write($writer)
	{
		if ($this->id!=null) $writer->writeValue(0, $this->id);
		$writer->writeString(1, $this->guid);
		$writer->writeString(2, $this->name);
	}
}
?>

File: Sitana.Framework/CrashReporter/Data/Application.cs
using System;
using Ebatianos.DataTransfer;

namespace Ebatianos.CrashReporter.Data
{
	public class Application: IEcsStructure
	{
		public int? Id;
		public string Guid;
		public string Name;

        void IEcsStructure.Read(EcsReader reader)
		{
			Id = reader.HasField(0) ? (int?)reader.ReadInt32(0) : null;
			Guid = reader.ReadString(1);
			Name = reader.ReadString(2);
		}

        void IEcsStructure.Write(EcsWriter writer)
		{
			if (Id!=null) writer.Write(0, Id.Value);
			writer.Write(1, Guid);
			writer.Write(2, Name);
		}
	}
}